connectors/sources/outlook.py

```diff
@@ -0,0 +1,944 @@
+            ],
+            "company_name": contact.company_name,
+            "birthday": ews_format_to_datetime(
+                source_datetime=contact.birthday, timezone=timezone
+            ),
+        }
+
+    def attachment_doc_formatter(self, attachment, timezone):
+        return {
+            "_id": attachment.attachment_id.id,
+            "title": attachment.name,
+            "type": "attachment",
+            "_timestamp": ews_format_to_datetime(
+                source_datetime=attachment.last_modified_time, timezone=timezone
+            ),
+            "size": attachment.size,
+        }
+
+
+class OutlookClient:
+    """Outlook client to handle API calls made to Outlook"""
+
+    def __init__(self, configuration):
+        self._sleeps = CancellableSleeps()
+        self.configuration = configuration
+        self._logger = logger
+        self.is_cloud = self.configuration["data_source"] == OUTLOOK_CLOUD
+
+    def set_logger(self, logger_):
+        self._logger = logger_
+
+    @cached_property
+    def _get_user_instance(self):
+        if self.is_cloud:
+            return Office365Users(
+                client_id=self.configuration["client_id"],
+                client_secret=self.configuration["client_secret"],
+                tenant_id=self.configuration["tenant_id"],
+            )
+
+        self.ssl_enabled = self.configuration["ssl_enabled"]
+        self.certificate = self.configuration["ssl_ca"]
+        if self.ssl_enabled and self.certificate:
+            self.ssl_ctx = ssl_context(certificate=self.certificate)
+        else:
+            self.ssl_ctx = False
+
+        return ExchangeUsers(
+            ad_server=self.configuration["active_directory_server"],
+            domain=self.configuration["domain"],
+            exchange_server=self.configuration["exchange_server"],
+            user=self.configuration["username"],
+            password=self.configuration["password"],
+            ssl_enabled=self.ssl_enabled,
+        )
+
+    async def ping(self):
+        await anext(self._get_user_instance.get_users())
+
+    async def get_mails(self, account):
+        for mail_type in MAIL_TYPES:
+            if mail_type["folder"] == "archive":
+                # If 'Archive' folder is not present, skipping the iteration
+                try:
+                    folder_object = (
+                        account.root / "Top of Information Store" / "Archive"
+                    )
+                except Exception:
+                    continue
+            else:
+                folder_object = getattr(account, mail_type["folder"])
+
+            for mail in folder_object.all().only(*MAIL_FIELDS):
+                yield mail, mail_type
+
+    async def get_calendars(self, account):
+        for calendar in account.calendar.all().only(*CALENDAR_FIELDS):
+            yield calendar
+
+    async def get_child_calendars(self, account):
+        for child_calendar in account.calendar.children:
```

Comment:
I didn't do any tests. Just want to make sure do you need to call something like `.all()` to make it iterate.

Reply:
No, this is another property of calendar object so there is no need of `.all()` to make it iterate.